version: '3.8'

services:
  db:
    image: postgres:17-alpine
    container_name: it_issue_tracker_db
    restart: always
    ports:
      - "5433:5432"
    env_file:
      - /.env
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - app-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"]
      interval: 10s
      timeout: 5s
      retries: 5

networks:
  app-network:
    driver: bridge

volumes:
  postgres_data:
    driver: local